src/verify_cert.rs

```diff
@@ -151,8 +151,7 @@ fn build_chain_inner(
             const MAX_SUB_CA_COUNT: usize = 6;
 
             if sub_ca_count >= MAX_SUB_CA_COUNT {
-                // TODO(XXX): Candidate for a more specific error - Error::PathTooDeep?
-                return Err(Error::UnknownIssuer);
+                return Err(Error::MaximumPathDepthExceeded);
```

Comment:
does this limit the length of certificate chain to 6?

Reply:
yes, though this error is not fatal so the search can continue to find valid shorter chains.

Reply:
Also, this branch only changes the error variant to be more specific and adds test coverage for this case. The limit being imposed is unchanged and inherited from briansmith/webpki.

Reply:
@stanal Do you have examples of real world certificate paths issued by a trust anchor participating in the web PKI that exceed this depth?

Reply:
> @stanal Do you have examples of real world certificate paths issued by a trust anchor participating in the web PKI that exceed this depth?
no, acturally , I cann't konw.  Although it does not exist for now, not means it won't exist in the future, may be will has one 7 or 8 ... and so on. I'm just not sure of that.

Reply:
I believe the limit of 6 comes from NSS libmozpkix -- eg https://github.com/nss-dev/nss/blob/master/lib/mozpkix/lib/pkixbuild.cpp#L322-L332 -- and therefore provides a bit of reassurance that people don't rely on certificate chains longer than this in the public internet.

Reply:
thanks! got it !

Reply:
As another data-point, while looking into something unrelated I noticed that Java+Android's [`PKIXBuilderParameters`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/security/cert/PKIXBuilderParameters.html) defaults to a maximum path length of 5.